### ISSUE TITLE
Clipboard/buffer fails daily....still...

### ISSUE THREAD

**Issue by iam8up** [synergy/synergy#5034]
I've had this issue for years.  Quite frankly it's simply getting tiresome.
https://github.com/synergy/synergy/issues/95
It was reported several times on SF over the years but those links appear to be gone.

I've discussed it a few times in IRC, no one can figure out what's causing it.  I should also mention with the donation I provided I was supposed to get the additional feature of copying files across machines - that never worked.  My only resolution is to restart the service on both the client and server.

For some time both machines were Windows 7.  Now, it is Windows 10 and Windows 7.  It doesn't appear to have changed the frequency of it happening.  I'm currently running 1.7.4 PRO.

Can we please fix this issue once and for all?  I'd be happy to help in any way I can.

**Comment by geoff10e** [comment]
I was coming here to submit a similar issue. Maybe the same. I found that I can copy and paste from server to client, and server to server and client to server. I can not copy from client to client unless I bring the mouse over to the server and then back to the client.

**Comment by gclayburg** [comment]
Ditto on that!  I see the same issue with version 1.7.4 with the server running on Centos linux and the client on Linux Mint.  The synergy server still fails many times per day and needs to be manually restarted.  The failure is always triggerred by trying to copy and paste clipboard content from the client to the server.  The issue is intermittant, but fairly easy to reproduce.  Is anyone looking at fixing this?  I could submit more detailed steps to reproduce it if it would help...

**Comment by iam8up** [comment]
Personally my issue has repeatedly been copying text from the server to
client.

**Comment by ctlajoie** [comment]
@gclayburg I had the same issues but recently I tried out the code in PR #4994 and I think it resolved all of them. At least I haven't noticed any problems yet. Anyway if you guys are able to compile from source, give it a try.

**Comment by nyetwurk** [comment]
I still have issues with server->client clipboard, even with my PR.

**Comment by ctlajoie** [comment]
I'm just thrilled that it doesn't segfault all the time tbh. It is odd that some people experience vastly different behavior than others when using this software, even taking into account that everyone uses it with different combinations of OSes.

**Comment by nyetwurk** [comment]
See #4735 and #5041

**Comment by iam8up** [comment]
OK so what the issue is reported in countless places, people report the
issue and after testing a nightly build without resolution only to end up
never fixing it?  I'm not sure what you're suggesting.

On Thu, Oct 1, 2015 at 6:45 PM, Nye Liu <notifications@github.com> wrote:

> See #4735 <https://github.com/synergy/synergy/issues/4735> and #5041
> <https://github.com/synergy/synergy/issues/5041>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/synergy/synergy/issues/5034#issuecomment-144868984>.
>



-- 
Josh Luthman

A patch in time saves kill -9
-http://www.tuxmaniac.com

**Comment by ctlajoie** [comment]
There does seem to be a lack of interest by the core devs in fixing these issues. Simply merging @nyetwurk's PR would be a start, since it at least it doesn't break anything, and at best it fixes several things including the clipboard for some of us.

**Comment by nyetwurk** [comment]
I've been trying to stamp out the clipboard bugs as best as I can in the bugs referenced. Didn't realize there was an irc channel for synergy. I will check it out.

So far my pull request has not been merged with upstream, but you're free to check out my branch and see if it fixes things for you.

FWIW I share your frustration. So far, there has been almost zero response from the primary developers to any of my questions about the source.

That said, there doesn't seem to be anybody but me working on these bugs either. Not sure why nobody is contributing any effort to this software. The source code is all there, it is relatively well thought out and pretty readable.

**Comment by iam8up** [comment]
I can certainly offer my assistance in that my workstation has this problem
very consistently.  I don't believe I would be of much (or any) help as a
developer.

I appreciate your effort in working on the issue, Nye!

On Fri, Oct 2, 2015 at 2:30 PM, Nye Liu <notifications@github.com> wrote:

> I've been trying to stamp out the clipboard bugs as best as I can in the
> bugs referenced. Didn't realize there was an irc channel for synergy. I
> will check it out.
>
> So far my pull request has not been merged with upstream, but you're free
> to check out my branch and see if it fixes things for you.
>
> FWIW I share your frustration. So far, there has been almost zero response
> from the primary developers to any of my questions about the source.
>
> That said, there doesn't seem to be anybody but me working on these bugs
> either. Not sure why nobody is contributing any effort to this software.
> The source code is all there, it is relatively well thought out and pretty
> readable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/synergy/synergy/issues/5034#issuecomment-145116554>.
>



-- 
Josh Luthman

A patch in time saves kill -9
-http://www.tuxmaniac.com

**Comment by crlsgms** [comment]
I just updated my installation of synergy, it was quite outdated since the versions I installed after 1.4.2 were mostly laggy on the client side, and clippboard stopped from nowhere. Now with 1.7.4 the disconnection only occurs if the client gets on its screensaver, or if I copy a big block of text on the client side. 

On pure text formats it was usable in the past, now not even copying the content to a pure text editor before the copy I can send data from the clients to the server. 

Using win7 64 as server, mint LMDE and debian 8 clients.
